server/services/docs.ts

```diff
@@ -0,0 +1,99 @@
+import { readdirSync, readFileSync, statSync } from 'fs'
+import matter from 'gray-matter'
+import { nanoid } from 'nanoid'
+import { DocFileOrHeading, DocsMenuContent } from '../../types/interfaces'
+
+type DocsFrontMatter = {
+  priority?: number
+}
+
+const sortByPriority = (a: DocFileOrHeading, b: DocFileOrHeading): number => {
+  // As the highest priority is 1, we don't care about 0
+  if (a.priority && b.priority) return a.priority - b.priority
+  if (!a.priority && b.priority) return 1
+  if (a.priority && !b.priority) return -1
+  return 0
+}
+
+const convertFileOrDirNameToTitle = (str: string): string =>
+  str
+    .split('-')
+    .map((word) => `${word[0].toUpperCase()}${word.slice(1)}`)
+    .join(' ')
+
+const removeFileExtension = (fileOrDirName: string): string => {
```

Comment:
Prefer standard library functions, in this case `Path.parse()` provides the required functionality. https://nodejs.org/api/path.html#pathparsepath

```
const { name } = parse(fileOrDirName)
```

Reply:
Done